src/app/components/wallet-widget/wallet-widget.component.ts

```diff
@@ -67,7 +69,24 @@ export class WalletWidgetComponent implements OnInit {
   }
 
   async unlockWallet() {
-    await new Promise(resolve => setTimeout(resolve, 500)); // brute force delay
+    if (this.shouldDelayUnlockAttempt === true) {
+      await new Promise(resolve => setTimeout(resolve, 500)); // brute force delay
```

Comment:
I think some additional logic must be set here. Since the function is async you can press the buttons and bruteforce how much you want. The delay has no affect? Maybe set a global variable that is reset after 500ms. And if that variable is set, just return the unlockWallet function. I made this wrong from the very beginning it seems.